### ISSUE TITLE
NullableToStringCall false positives with custom contract-enforced function

### ISSUE THREAD

**Issue by brentwatson** [detekt/detekt#9145]
## Observed Behavior

I have the following custom function which uses kotlin contracts -- essentially just the inverse of the kotlin stdlib `isNullOrBlank` extension function:

```kotlin
@OptIn(ExperimentalContracts::class)
fun CharSequence?.isNotNullOrBlank(): Boolean {
    contract {
        returns(true) implies (this@isNotNullOrBlank != null)
    }
    return !this.isNullOrBlank()
}
```

When I used it in Kotlin 2.2 detekt correctly didn't warn about `NullableToStringCall`. Now it does when using Kotlin 2.3:

```kotlin
data class Demo(
    val firstName: String?,
    val lastName: String?,
    val default: String,
) {
    fun getDisplayName(): String {
        return if (firstName.isNotNullOrBlank() && lastName.isNotNullOrBlank()) {
            "$firstName $lastName" // Detekt report NullableToStringCall here
        } else default
    }
}
```

## Your Environment

Version of detekt used: 1.23.8
Version of Gradle used: 9.2.1
Version of Kotlin used: 2.3.10
Operating System and version: MacOS Tahoe 26.1

**Comment by atulgpt** [comment]
Hi @brentwatson I have added TCs in the [PR](https://github.com/detekt/detekt/pull/9148) for your case, but in my case detekt working fine
Since contract case was missing from the existing TCs we can merge this PR

**Comment by brentwatson** [comment]
@atulgpt Do these tests pass on latest stable release (`1.23.8`) with Kotlin 2.3? Or is support for 2.3 only planned for detekt 2.0+? I tried the tests on `1.23.8` tag but can't seem to change kotlin target to 2.3 without additional project errors.

**Comment by BraisGabin** [comment]
There is no plan to support Kotlin 2.3+ in detekt 1.x. I recommend you to move to detekt 2.x

**Comment by brentwatson** [comment]
The few issues I opened seem to be happening with Detekt 1.23.8 + Kotlin 2.3. I'll close them and start migration to Detekt 2.0
